### PR TITLE
feat: support setting hostname to the machine name

### LINKED ISSUE #131: allow hostnames to be set to the machine name (TalosConfig spec field)

### PR DESCRIPTION
**Pull request by smira**
Fixes #131

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ Fields available in the `TalosConfigTemplate` (and `TalosConfig`) resources:
    It is recommended to always set this field explicitly to avoid issues when CABPT is upgraded to the version which supports new Talos version.
 - `configPatches` (optional): set of machine configuration patches to apply to the generated configuration.
 - `data` (only for `generateType: none`): user-supplied machine configuration.
+- `hostnameFromMachineName`: set the hostname in the generated machine configuration to the `Machine` name (not supported with `MachinePool` deployments)
```
Comment (andrewrynhard) [inline README.md:89]: What about something like "GenerateHostname" and keeping it generic?

Comment (smira) [reply README.md:89]: I thought about something like

```yaml
spec:
   hostname:
      source: machineName
```

?

**Comment by smira** [comment]
I went all the way and added validation webhooks and full support for them in the integration testing. We can do more validation and we should do that in the future!

File: README.md (hunk 1 above)
Comment (andrewrynhard) [reply README.md]: I like this idea. We can expand it in different ways.

**Review by andrewrynhard — APPROVED** [review]

**Comment by smira** [comment]
I will add health checks in the follow-up PR.

**Pull request merged**